sdgx/data_models/inspectors/numeric.py

```diff
@@ -14,69 +14,132 @@ class NumericInspector(Inspector):
+        int_cnt = (numeric_values == numeric_values.astype(int)).sum()
+
+        # Calculate the ratio of integer values to the total numeric values
+        int_rate = int_cnt / len(numeric_values)
+
+        # Return True if the integer rate is greater than the predefined threshold
+        return int_rate > self._int_rate
+
+    def _is_positive_or_negative_column(
+        self, col_series: pd.Series, threshold: float, comparison_func
+    ) -> bool:
+        """
+        Determine if a column contains predominantly positive or negative values.
+
+        This method checks if the proportion of values that satisfy a given comparison
+        function exceeds a predefined threshold.
+
+        Args:
+            col_series (pd.Series): The column series to be inspected.
+            threshold (float): The proportion threshold for considering the column as positive or negative.
+            comparison_func (function): A function that takes a numeric value and returns a boolean.
+
+        Returns:
+            bool: True if the column satisfies the condition, False otherwise.
+        """
+        # Convert the column series to numeric values, coercing errors to NaN and dropping NaN values
+        numeric_values = pd.to_numeric(col_series, errors="coerce").dropna()
+
+        # Apply the comparison function to the numeric values and sum the results
+        count = comparison_func(numeric_values).sum()
+
+        # Calculate the proportion of values that meet the comparison criteria
+        proportion = count / len(numeric_values)
+
+        # Return True if the proportion meets or exceeds the threshold, otherwise False
+        return proportion >= threshold
+
+    def _is_positive_column(self, col_series: pd.Series) -> bool:
+        """
+        Determine if a column contains predominantly positive values.
+
+        This method checks if the proportion of positive values in the given column
+        exceeds a predefined threshold.
+
+        Args:
+            col_series (pd.Series): The column series to be inspected.
+
+        Returns:
+            bool: True if the column is predominantly positive, False otherwise.
+        """
+        return self._is_positive_or_negative_column(col_series, self.pos_threshold, lambda x: x > 0)
+
+    def _is_negative_column(self, col_series: pd.Series) -> bool:
+        """
+        Determine if a column contains predominantly negative values.
+
+        This method checks if the proportion of negative values in the given column
+        exceeds a predefined threshold.
+
+        Args:
+            col_series (pd.Series): The column series to be inspected.
+
+        Returns:
+            bool: True if the column is predominantly negative, False otherwise.
+        """
+        return self._is_positive_or_negative_column(
+            col_series, self.negative_threshold, lambda x: x < 0
+        )
 
     def fit(self, raw_data: pd.DataFrame, *args, **kwargs):
```

Comment:
Both two `for` loop can merge to one, small refer shoot:

```python
def fit(self, raw_data: pd.DataFrame, *args, **kwargs):
    self.int_columns = set()
    self.float_columns = set()
    self.positive_columns = set()
    self.negative_columns = set()
    self.df_length = len(raw_data)
    
    for col in raw_data.columns:
        if raw_data[col].dtype in ['int64', 'float64']:
            if self._is_int_column(raw_data[col]):
                self.int_columns.add(col)
            else:
                self.float_columns.add(col)
            
            if self._is_positive_column(raw_data[col]):
                self.positive_columns.add(col)
            elif self._is_negative_column(raw_data[col]):
                self.negative_columns.add(col)
    
    self.ready = True

```

Reply:
I have updated the reviewed content in the code for this PR.